### PR TITLE
feat(log): print failed changesets to the default UI

### LINKED ISSUE #1279: Liquibase should have a better logging with failOnError=false when error occurs
## Description
Liquibase changeSets have an attribute "failOnError".   When set to "false", Liquibase will not fail execution and continue on to the next changeSet if the current one is failing.
This is the expected behavior.
However, in the event where some failing changeSets with "failOnError=false" during deployment, Liquibase will just output "_Liquibase: Update has been successful._" 

It will be more helpful to troubleshoot if Liquibase had a better output/logging specifying the errors during deployment. 

## Steps To Reproduce
1. Create a changeSet with failOnError:false and a bad syntax:

```
--liquibase formatted sql

--changeset SteveZ:45555-createTable_TABLE200 context:UAT labels:Jira-166 failOnError:false
CREATE TABLE TABLE200 
  (
    JOB_ID VARCHAR2(10 BYTE) NOT NULL, 
    MIN_SALARY NUMBERed(6, 0), 
    MAX_SALARY NUMBER(6, 0), 
    CONSTRAINT JOB_ID_PK200 PRIMARY KEY (JOB_ID)
  )
;
--rollback drop table TABLE200
```

1. run: liquibase update

## Actual Behavior
Liquibase: Update has been successful.

## Expected/Desired Behavior
It would be helpful is Liquibase had outputted/logged more information about what errors happened during deployment.

## Additional Context
Things can get even more complicated when there is a script with multiple statement being deployed, where some of the statements are successful and some are not.

For example, consider the following script:

```
 --changeset SteveZ:45555-createTable_TABLE200and201 context:UAT labels:Jira-166 failOnError:false
CREATE TABLE TABLE200 
  (
    JOB_ID VARCHAR2(10 BYTE) NOT NULL, 
    MIN_SALARY NUMBER(6, 0), 
    MAX_SALARY NUMBER(6, 0), 
    CONSTRAINT JOB_ID_PK200 PRIMARY KEY (JOB_ID)
  )
;

CREATE TABLE TABLE201 
  (
    JOB_ID VARCHAR2(10 BYTE) NOT NULL, 
    MIN_SALARY NUMBERed(6, 0), 
    MAX_SALARY NUMBER(6, 0), 
    CONSTRAINT JOB_ID_PK201 PRIMARY KEY (JOB_ID)
  )
;
```

Here there are two statements in one changeSet with the failOnError=false attribute.
When running: liquibase update

1. Liquibase will run this changeSet
1. The first statement will create the table "TABLE200"
1. The second statement will fail because of "NUMBERed" bad syntax.
1. Liquibase will fail on this changeSet and will NOT create a row in the DATABASECHANGELOG
This will result in an unexpected object TABLE200 in the database.
If Liquibase had a better logging, then the user could at least had seen what happened during the deployment.

## Acceptance Criteria
1. We need to respect the optional failOnError setting, but change the silent fail. That is, they do not need to stop on error (thus their optional setting) but still need to know on error
1. This notification can be missed in automation, but we can still make the information available to the user.
1. The following message should be output to the log at WARNING level

```java
TBD
```

### PR DESCRIPTION
**Pull request by filipelautert**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [X] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
feat(log): print failed changesets to the default UI, so when a changeset fails but liquibase reaches the end of  normal execution, the user is made aware of the failure. 

## Things to be aware of
Summary has a similar feature including more details and a formatted description.

## Additional Context
Fixes #1279

The new output looks like:

```
Running Changeset: changelog.xml::1710355418505-1::filipe
Running Changeset: changelog.xml::1710355418505-2::filipe
Running Changeset: changelog.xml::1710355418505-3::filipe

Errors encountered while deploying the following changesets, for more information use the --log-level flag: 
     changelog.xml::1710355418505-2::filipe
     changelog.xml::1710355418505-3::filipe

Liquibase: Update has been successful. Rows affected: 1
Liquibase command 'update' was executed successfully.
```

### CODE REVIEW

File: liquibase-standard/src/main/java/liquibase/command/core/UpdateCountCommandStep.java [hunk 1]
```diff
@@ -115,11 +111,7 @@ protected void customMdcLogging(CommandScope commandScope) {
     }
 
     @Override
-    public void postUpdateLog(int rowsAffected) {
-        if (rowsAffected > -1) {
-            Scope.getCurrentScope().getUI().sendMessage(String.format(coreBundle.getString("update.count.successful.with.row.count"), rowsAffected));
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/command/core/UpdateCountCommandStep.java:120]: Update-count has been replaced to Update has been successful as noticed by @rberezen

File: liquibase-standard/src/main/java/liquibase/command/core/UpdateToTagCommandStep.java [hunk 2]
```diff
@@ -174,11 +174,7 @@ protected void customMdcLogging(CommandScope commandScope) {
     }
 
     @Override
-    public void postUpdateLog(int rowsAffected) {
-        if (rowsAffected > -1) {
-            Scope.getCurrentScope().getUI().sendMessage(String.format(coreBundle.getString("update.to.tag.successful.with.row.count"), rowsAffected));
```
Comment (filipelautert) [inline liquibase-standard/src/main/java/liquibase/command/core/UpdateToTagCommandStep.java:179]: Update-to-tag has been replaced to Update has been successful as noticed by @rberezen

File: liquibase-standard/src/main/java/liquibase/command/core/UpdateCountCommandStep.java (hunk 1 above)
Comment (filipelautert) [reply liquibase-standard/src/main/java/liquibase/command/core/UpdateCountCommandStep.java:120]: @rberezen fixed!

**Review by rberezen — APPROVED** [review]

File: liquibase-standard/src/main/java/liquibase/command/core/AbstractUpdateCommandStep.java [hunk 3]
```diff
@@ -283,12 +283,42 @@ public void setFastCheckEnabled(boolean fastCheckEnabled) {
...
+    }
 
+    /**
+     * Generic method called by all Update commands that actually apply the change to the database (ie !update-sql)
+     * @param rowsAffected
+     * @param exceptionChangeSets
+     */
+    protected void postUpdateLogForActualUpdate(int rowsAffected, List<ChangeSet> exceptionChangeSets) {
+        if (exceptionChangeSets != null && !exceptionChangeSets.isEmpty()) {
+            Scope.getCurrentScope().getUI().sendMessage("Errors encountered while deploying the following changesets, for more information use the --log-level flag: ");
```
[the remark this answers is not in the document]
Comment (lbadam) [reply liquibase-standard/src/main/java/liquibase/command/core/AbstractUpdateCommandStep.java:304]: Agreed 👍

[the remark this answers is not in the document]
Comment (filipelautert) [reply liquibase-standard/src/main/java/liquibase/command/core/AbstractUpdateCommandStep.java:304]: Sure!
```
UPDATE SUMMARY
Run:                          1
Previously run:               0
Filtered out:                 0
Failed deployment:            2
-------------------------------
Total change sets:            3

Errors encountered while deploying the following changesets: 
     changelog.xml::1710355418505-2::filipe
     changelog.xml::1710355418505-3::filipe
For more information use the --log-level flag.

Liquibase: Update has been successful. Rows affected: 1
```

Better now?

**Review by tati-qalified — APPROVED** [review]

**Pull request merged**